packages/components/src/components/flyout/flyout.tsx

```diff
@@ -189,51 +175,78 @@ export class Flyout {
             <slot />
           </div>
           {this.hasFooter && (
-            <div class="footer" ref={(el) => (this.footer = el)}>
+            <div key="footer" class="footer" ref={(el) => (this.footer = el)}>
               <slot name="footer" />
             </div>
           )}
           {this.hasSubFooter && (
-            <div class="sub-footer" ref={(el) => (this.subFooter = el)}>
+            <div key="sub-footer" class="sub-footer" ref={(el) => (this.subFooter = el)}>
               <slot name="sub-footer" />
             </div>
           )}
         </div>
-      </Host>
+      </dialog>
     );
   }
 
-  private onMouseDown = (e: MouseEvent): void => {
-    if ((e.composedPath() as HTMLElement[])[0] === this.host && !clickStartedInScrollbarTrack(this.host, e)) {
-      this.dismissFlyout();
-    }
+  private updateHeaderShadow = (): void => {
+    const shouldApplyShadow = this.wrapper.scrollTop > FLYOUT_SCROLL_SHADOW_THRESHOLD;
+
+    this.header.classList.toggle(headerShadowClass, shouldApplyShadow);
   };
 
-  private updateFocusTrap = (isOpen: boolean): void => {
-    setFocusTrap(this.host, isOpen, this.dialog, this.dismissBtn, this.dismissFlyout);
+  private updateFooterShadow = (): void => {
+    const shouldApplyShadow = this.subFooter.offsetTop > this.wrapper.clientHeight + this.wrapper.scrollTop;
+
+    this.footer.classList.toggle(footerShadowClass, shouldApplyShadow);
   };
 
   // eslint-disable-next-line @typescript-eslint/member-ordering
-  private onScroll = throttle(100, () => {
-    if (this.dialog.scrollHeight > this.dialog.clientHeight) {
+  private updateShadow = throttle(100, () => {
+    if (this.wrapper.scrollHeight > this.wrapper.clientHeight) {
       this.updateHeaderShadow();
+
       if (this.hasFooter) {
         this.updateFooterShadow();
       }
     }
   });
 
-  private updateHeaderShadow = (): void => {
-    const shouldApplyShadow = this.dialog.scrollTop > FLYOUT_SCROLL_SHADOW_THRESHOLD;
-    this.header.classList.toggle(headerShadowClass, shouldApplyShadow);
+  private onClickDialog = (e: MouseEvent & { target: HTMLElement }): void => {
+    if (e.target.tagName === 'DIALOG') {
+      // dismiss dialog when clicked on backdrop
+      this.dismissDialog();
+    }
   };
 
-  private updateFooterShadow = (): void => {
-    const shouldApplyShadow = this.subFooter.offsetTop > this.dialog.clientHeight + this.dialog.scrollTop;
-    this.footer.classList.toggle(footerShadowClass, shouldApplyShadow);
+  private onSlotChange = (): void => {
+    forceUpdate(this.host);
+
+    this.dismissBtn.focus();
   };
 
-  private dismissFlyout = (): void => {
+  private onCancelDialog = (e: Event): void => {
+    // prevent closing the dialog uncontrolled by ESC (only relevant for browsers supporting <dialog/>)
+    e.preventDefault();
+
+    this.dismissDialog();
+  };
+
+  private setDialogVisibility(isOpen: boolean): void {
+    if (!this.dialog) {
+      return;
+    }
```

Comment:
We shouldn't need this especially when the reason for adding this is a failing unit test.